lib/download.js

```diff
@@ -86,13 +89,28 @@ module.exports = function (url, version, dest, cb) {
       }
     });
 
-    src.pipe(gunzip).pipe(tar.Extract({ path: dest }))
-      .on('end',function () {
+    src.pipe(gunzip).pipe(tar.Extract({path: dest}))
+      .on('end', function() {
         console.log(url + ' is now extracted to ' + dest);
-        cb && cb();
-      }).on('error', function (error) {
-        cb && cb(error);
-      });
+        var err;
+        var home = homedir();
+        var icDir = path.join(home, 'oracle-instant-client');
+        if (!fs.existsSync(icDir)) {
+          try {
+            fse.copySync(path.join(dest, 'instantclient'), icDir,
```

Comment:
Is there a reason not to do this check before downloading and just extracting the tar stream directly into the destination if it doesn't already exist?